packages/shared/lib/services/connection.service.ts

```diff
@@ -1742,6 +1742,90 @@ class ConnectionService {
         return Err(new NangoError('failed_to_get_connections_count'));
     }
 
+    async billableConnections(referenceDate: Date): Promise<
+        Result<
+            {
+                accountId: number;
+                count: number;
+                year: number;
+                month: number;
+            }[],
+            NangoError
+        >
+    > {
+        // Note:
+        // a billable connection is a connection that is not deleted and has not been deleted during the month
+        // connections are pro-rated based on the number of seconds they were existing in the month
+
+        const targetDate = new Date(Date.UTC(referenceDate.getUTCFullYear(), referenceDate.getUTCMonth(), referenceDate.getUTCDate(), 0, 0, 0, 0));
+        const year = referenceDate.getUTCFullYear();
+        const month = referenceDate.getUTCMonth() + 1; // js months are 0-based
+
+        const res = await db.readOnly
+            .with('month_info', (qb) => {
+                qb.select(
+                    db.readOnly.raw(`DATE_TRUNC('month', ?::date) AS month_start`, [targetDate]),
+                    db.readOnly.raw(`(DATE_TRUNC('month', ?::date) + INTERVAL '1 month' - INTERVAL '1 day')::date AS month_end`, [targetDate]),
+                    db.readOnly.raw(
+                        `EXTRACT(EPOCH FROM (DATE_TRUNC('month', ?::date) + INTERVAL '1 month') - DATE_TRUNC('month', ?::date)) AS total_seconds_in_month`,
+                        [targetDate, targetDate]
+                    )
+                );
+            })
+            .with('billable_connections', (qb) => {
+                qb.select(
+                    'e.account_id',
+                    'c.id as connection_id',
+                    'c.created_at',
+                    db.readOnly.raw(`COALESCE(c.deleted_at, (SELECT month_end FROM month_info) + INTERVAL '1 day') AS effective_end_date`),
+                    db.readOnly.raw(`(SELECT month_start FROM month_info) AS month_start`),
+                    db.readOnly.raw(`(SELECT month_end FROM month_info) AS month_end`),
+                    db.readOnly.raw(`(SELECT total_seconds_in_month FROM month_info) AS total_seconds_in_month`)
+                )
+                    .from('nango._nango_connections as c')
+                    .join('nango._nango_environments as e', 'c.environment_id', 'e.id')
```

Comment:
does ALL the connections for ALL environments count for billing? @bastienbeurier

Reply:
Yes!

Reply:
I thought dev env was not billed?